pkg/tfgen/generate_test.go

```diff
@@ -132,3 +136,112 @@ func Test_GenerateTestDataSchemas(t *testing.T) {
 		bridgetesting.AssertEqualsJSONFile(t, schemaPath, schema)
 	}
 }
+
+// Encoding of Elem() and Type() is tricky when recognizing types, cover all the cases here documented in the
+// shim.Schema.Elem() docstring.
+func Test_makePropertyType(t *testing.T) {
```

Comment:
@guineveresaenger thanks for insisting on proper unit tests cases here, here we go. Covering the interesting bits.